zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/BpmnStreamProcessor.java

```diff
@@ -171,6 +171,9 @@ private void processEvent(
         break;
       case TERMINATE_ELEMENT:
         final var terminatingContext = stateTransitionBehavior.transitionToTerminating(context);
+        // cancel any active job associated with the element being terminated
+        // (e.g. execution listener, serviceTask job)
+        jobBehavior.cancelJob(terminatingContext);
         processor.onTerminate(element, terminatingContext);
```

Comment:
:wrench: Assuming that we keep the canceling behavior in the processors, we should flip the invocation and invoke the processor first to cancel the (service task) job within the processor. 

```suggestion
        processor.onTerminate(element, terminatingContext);
        // cancel any job associated with the element being terminated (e.g. execution listener)
        jobBehavior.cancelJob(terminatingContext);
```

Reply:
🤔 After testing the suggested approach, I encountered an issue: invoking the job cancellation method after `processor.onTerminate` results in the following error:

```
Cannot invoke "io.camunda.zeebe.engine.state.instance.ElementInstance.getJobKey()" because "elementInstance" is null
```

This happens because `elementInstance` is deleted inside `ProcessInstanceElementTerminatedApplier#applyState`, which is triggered by `stateTransitionBehavior.transitionToTerminated` within the `processor.onTerminate` method.

Also, currently, it’s not possible to have multiple active jobs for an element; we can only have one—either a job associated with the element (e.g., a service task) or an EL job. 
We also clear `ElementInstance#jobKey` after each job completion.

---
Given these findings, @saig0 which approach would you prefer moving forward?
1. Adding `jobBehavior.cancelJob(context)` explicitly in each processor for elements that support ELs to maintain consistency.
2. Keeping the current implementation where the job canceling behavior is handled as it is now, centralised inside  `BpmnStreamProcessor`.